### DISCUSSION TITLE
Why can't I load the local font?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10959] [Q&A]
Firstly, I don't want to embed the font into the program for I want the user to choose the font he wants in my program. Then my program will automatically download the font into a directory and use it.

What's more, I don't want the font register into the system, for example place it into C:\windows\fonts\

I made a minimal sample project. Here is the font that I want to load:https://fonts.google.com/specimen/Josefin+Sans

I set Copy if newer to all fonts:
![image](https://user-images.githubusercontent.com/30720036/230744785-bd96e50f-6857-4914-839c-d57d963f1afa.png)

Here is XAML:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
        Title="AvaloniaApplication1">
	<TextBlock Name="TB">Whereas recognition of the inherent dignity</TextBlock>
</Window>
```
And here is code-behind:
```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Media;
using System;
using System.IO;

namespace AvaloniaApplication1
{
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
            if (File.Exists(System.AppDomain.CurrentDomain.BaseDirectory + @"\Fonts\JosefinSans-Regular.ttf"))
            {
                FontFamily FF = new FontFamily(new Uri(System.AppDomain.CurrentDomain.BaseDirectory + @"\Fonts\#Josefin Sans", UriKind.Absolute), "Josefin Sans");
                TB.Text = TB.Text + " FontFamily Changed";
                TB.FontFamily = FF;
            }
        }
    }
}
```

Here is the font should be like:
![image](https://user-images.githubusercontent.com/30720036/230744839-dd950577-4d77-4ed9-a556-13d757ad3bd5.png)
And here is my sample actually is:
![image](https://user-images.githubusercontent.com/30720036/230744854-ea114340-323e-40f1-bcb4-683f52e4cfb0.png)

VS doesn't report any error yet. What's wrong with my code?

**Comment by maxkatz6** [comment]
Check this documentation page. It also has examples for fonts:
https://docs.avaloniaui.net/docs/getting-started/assets#including-assets

In short, FontFamily constructor accepts embedded avalonia resource files, not the file system. So you need to use "avares://' scheme for Uri and properly include files.
